src/NuGet.Core/NuGet.CommandLine.XPlat/XPlatUtility.cs

```diff
@@ -0,0 +1,50 @@
+﻿using System;
+using System.Net;
+using Microsoft.Dnx.Runtime.Common.CommandLine;
+using NuGet.Common;
+using NuGet.Logging;
+using NuGet.Protocol.Core.Types;
+
+namespace NuGet.CommandLine.XPlat
+{
+    public class XPlatUtility
+    {
+        internal const string HelpOption = "-h|--help";
+        internal const string VerbosityOption = "-v|--verbosity <verbosity>";
+        internal static readonly int MaxDegreesOfConcurrency = Environment.ProcessorCount;
+
+        internal static LogLevel GetLogLevel(CommandOption verbosity)
+        {
+            LogLevel level;
+            if (!Enum.TryParse(value: verbosity.Value(), ignoreCase: true, result: out level))
+            {
+                level = LogLevel.Information;
+            }
+
+            return level;
+        }
+
+        public static void SetUserAgent()
```

Comment:
This might be breaking our contract of being able to override the user agent from an environment variable for testing purposes, can you validate if that is the case or not?